### PR TITLE
packages: add neuron kmod for 6.1 kernel

### PR DESCRIPTION
**Pull request by bcressey**
**Issue number:**
Related: #1995, #1075

**Description of changes:**
Package the Neuron kernel module for use on Inferentia and Trainium instance types.

**Testing done:**
Built `aws-dev` with local changes to include the new package, and a container with `neuron-ls` installed.

```
$ docker run -it --rm --device=/dev/neuron0 863599026182.dkr.ecr.us-west-2.amazonaws.com/neuron-tools:latest neuron-ls
Unable to find image '863599026182.dkr.ecr.us-west-2.amazonaws.com/neuron-tools:latest' locallym/neuron-tools:latest neuron-ls
latest: Pulling from neuron-tools
...
Status: Downloaded newer image for 863599026182.dkr.ecr.us-west-2.amazonaws.com/neuron-tools:latest
instance-type: inf1.2xlarge
instance-id: i-02d893416f9f374ce
+--------+--------+--------+---------+
| NEURON | NEURON | NEURON |   PCI   |
| DEVICE | CORES  | MEMORY |   BDF   |
+--------+--------+--------+---------+
| 0      | 4      | 8 GB   | 00:1f.0 |
+--------+--------+--------+---------+
```

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

File: packages/kmod-6.1-neuron/neuron-systemd-modules-load.drop-in.conf [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[Service]
+ExecStartPre=-/usr/bin/depmod
```
Comment (bcressey) [inline packages/kmod-6.1-neuron/neuron-systemd-modules-load.drop-in.conf:2]: This won't be necessary after we update to a `twoliter` release with this change:
https://github.com/bottlerocket-os/twoliter/pull/205

File: packages/kmod-6.1-neuron/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+[lib]
+path = "../packages.rs"
+
+[package.metadata.build-package]
+package-name = "kmod-6.1-neuron"
+releases-url = "https://awsdocs-neuron.readthedocs-hosted.com/en/latest/release-notes/runtime/aws-neuronx-dkms/index.html"
+
+[[package.metadata.build-package.external-files]]
+url = "https://yum.repos.neuron.amazonaws.com/aws-neuronx-dkms-2.16.7.0.noarch.rpm"
+sha512 = "8e66feb4051af31321c08b6663a950172da65c4e5b432c0b5609785be34ccb193c0eb50c9aadfeec8b6410ccbe05264a3fb6fc7cb66dc87b172bc5be5c4d92d0"
```
Comment (monirul) [inline packages/kmod-6.1-neuron/Cargo.toml:17]: Is this published in the amazon official sites or we generated it from the downloaded binary?

**Review by monirul — APPROVED** [review]
LGTM

**Review by arnaldo2792 — APPROVED** [review]

File: packages/kmod-6.1-neuron/Cargo.toml (hunk 2 above)
Comment (bcressey) [reply packages/kmod-6.1-neuron/Cargo.toml:17]: This is the offical site per the [documentation](https://awsdocs-neuron.readthedocs-hosted.com/en/latest/general/setup/neuron-setup/pytorch/neuronx/amazon-linux/torch-neuronx-al2023.html#setup-torch-neuronx-al2023).

**Pull request merged**